### ISSUE TITLE
Investigate the use of portable_simd

### ISSUE THREAD

**Issue by tlepoint** [tlepoint/fhe.rs#129]
Rust has an experimental [portable SIMD abstraction](https://doc.rust-lang.org/std/simd/index.html) which could be particularly useful to speed up some of the vector modular operations / NTTs in a "generic" way (i.e., without needing to specialize for AVX2 or Neon). I did some minimal investigation in the branch `simd-2`, and observe some speedup in a virtual machine with Intel CPU when enabling `avx2` and `avx512f` in `RUSTFLAGS`. Such a feature may require a lot of care to properly handle alignment, and this issue is created to centralize the discussions about SIMD.

**Comment by fionser** [comment]
How about the [ntt libray from Zama](https://github.com/zama-ai/concrete-ntt ) ?

**Comment by tlepoint** [comment]
Yes! Now that the latest bug in concrete-ntt is fixed, it should be easy to integrate it. I had a local branch with this integration a few weeks ago and need to revive it, but I also need to solve #229 before that.
